src/v/kafka/server/snc_quota_manager.cc

```diff
@@ -297,10 +298,103 @@ void snc_quota_manager::maybe_arm_balancer_timer() {
     _balancer_timer.arm(arm_until);
 }
 
+using shard_count_t
+  = boost::uint_t<std::numeric_limits<ss::shard_id>::digits>::fast;
+
+struct borrower_t {
+    ss::shard_id shard_id;
+    // can be 0 or 1, made shard_count_t for easier aggregation
+    inoutpair<shard_count_t> is_borrower;
```

Comment:
what is `boost::uint_t<std::numeric_limits<ss::shard_id>::digits>::fast;`? i think we should try to remove this. based solely on the type name `shard_count` an `unsigned` should work if you need it to match ss::shard_id.

Reply:
that basically means "unsigned type that is capable to hold `ss::shard_id`". I didn't want to use `ss::shard_id` as a shard count (it may be the same type but the domain is different), nor I wanted to hardcode a type based on what `shard_id` currently is. But maybe you're right that's an overkill and hardcoding that is good enough.